File: Dataframes/scala/rdd_a_dataframe.scala
//> using scala "3.6.3"
//> using repository "central"
//> using dep "org.apache.spark::spark-sql:3.5.4"

import org.apache.spark.sql.SparkSession

@main def connectToSpark(): Unit = {
  // Configurar la sesión de Spark
  val spark = SparkSession.builder()
    .appName("SparkConnectionTest")
    .master("local[*]")
    .getOrCreate()

  try {
    // Probar la conexión imprimiendo la versión de Spark
    println(s"Conexión establecida. Versión de Spark: ${spark.version}")

    // Data
    val data = Seq(("Jinx", 22), ("Solid Snake", 26), ("Naka", 39), ("Marlon", 21),
            ("Jhon Snow", 25), ("Solid Snake", 24), ("Marlon", 19))

    val dataDF = spark.createDataFrame(data).toDF("name", "age")

    val avgDF = dataDF.groupBy("name").agg(avg("age"))
    avgDF.show()
  } catch {
    case e: Exception => 
      println(s"Error al conectar con Spark: ${e.getMessage}")
  } finally {
    // Cerrar la sesión de Spark
    spark.stop()
  }
}

File: Dataframes/python/rdd_a_dataframe.py
from pyspark.sql import SparkSession
from pyspark.sql.functions import avg

# Creamos el DataFrame usando la SparkSession
spark = (SparkSession
  .builder
  .appName("PythonDataFrames")
  .getOrCreate())

# Creamos el DataFrame 
data_df = spark.createDataFrame([("Jinx", 22), ("Solid Snake", 26), ("Naka", 39),
    ("Marlon", 21), ("Jhon Snow", 25), ("Solid Snake", 24), ("Marlon", 19)], ["name", "age"])
# Agrupamos por nombre y aplicamos la agregación de la media de la edad (Age)
avg_df = data_df.groupBy("name").agg(avg("age"))
# Mostramos el resultado
avg_df.show()